### ISSUE TITLE
Early documentation makes assumptions about directive imports

### ISSUE THREAD

**Issue by underscorefunk** [Polymer/lit-html#1133]
I am stepping through the lit-html docs page by page to learn how it all works. I suspect many others are doing the same and appreciate the uncluttered approach of lit-html. I found a few spots that left me asking, "Why does this example not work."

## The problem: Location of directives are unclear in early documentation pages

In the [Writing Templates](https://lit-html.polymer-project.org/guide/writing-templates) page there are references made to _directives_ like repeat and cache. The same references are made on Styling Templates to _directives_ like styleMap and classMap. It's only through following the 'more on style map' link that I discovered where the directives are located and how to include them. Eliminating these points of friction will help people progress through the documentation more effortlessly. 

## Proposed solution: Update code block examples to include import statements for directives

Add the import statements to the code block examples on [Writing Templates](https://lit-html.polymer-project.org/guide/writing-templates) and [Styling Templates](https://lit-html.polymer-project.org/guide/styling-templates) so that they are more clear, consistent with entries on the [Template Syntax Reference: classMap](https://lit-html.polymer-project.org/guide/template-reference#classmap), etc

## Other recommendations (or for discussion)

1. [Writing Templates: Bind to Properties](https://lit-html.polymer-project.org/guide/writing-templates) should use standard html elements instead of custom elements to avoid confusion. Readers at this stage are still getting acclimated to lit-html. something like <input .value=${data.value} > is more clear. Perhaps add the native component example as a preceding example to the custom element example that exists for <my-list>

2. [Writing Templates: Nothing and the slot fallback content](https://lit-html.polymer-project.org/guide/writing-templates) makes many assumptions about reader knowledge early on. At this point in the docs there is no introduction about what a slot is, how html`<slot>...</slot>` relates to the <shadow-element> or how that fallback is pulled in when nothing is used in the example. 

3. Place "styleMap" and "classMap" at the beginning of the [Styling templates](https://lit-html.polymer-project.org/guide/styling-templates) page because they are the most simple to implement and require the least knowledge. Starting with things that look more like authoring stock html is easier for on-boarding. Complexity can build over time. Since lit-html is about authoring html (not web components) I believe it should start with concerns about simple html first, and then move to its integrations and implications with web components, shadow DOM, polyfills, etc.

**Comment by underscorefunk** [comment]
Also, maybe move 'concepts' in the navigation to above 'getting started'. 

In the concepts page it states 

> lit-html utilizes some unique properties of JavaScript template literals and HTML <template> elements to function and achieve fast performance. So it's helpful to understand them first.

... which is true, but you only really see this after the template syntax and other detailed documentation.

**Comment by justinfagnani** [comment]
Good catch @underscorefunk.

cc @arthurevans

**Comment by arthurevans** [comment]
Great feedback, thanks! I think most of your points are spot-on and easily implemented, as well.

Two things I have questions/comments on:

I'm not sure just moving the concepts section would be super helpful, because it has a bunch of implementation detail that I don't think folks need to know up front. Which parts in particular did you feel would be helpful? Just more up-front discussion of template literals, or were there other points in here that you really wished were explained up front?

"nothing and the slot fallback content" ... It conceptually fits here, but it's definitely advanced content that many people will never need. I wonder whether just putting a note to that effect at the front of that section would help? 

OK, it could also use some edits, because it's pretty darn cryptic as it stands.

**Comment by underscorefunk** [comment]
Thanks for the kind words @arthurevans . I'm glad you're finding my feedback helpful. 

In response to your questions:

**re: Moving concepts to after Getting Started and before Writing Templates**
"Concepts" covers what tagged templates are, how to create them, and the how to render them. It feels like an overview or intro to writing templates, rendering templates, etc. "Writing Templates" has similar examples to the concepts page, but without the more broad scope and with even more implementation details and complexity (conditionals, repeats, bindings, etc). Conceptually (pardon the pun) it feels like introducing the concepts should come before implementation. Writing templates, styling templates, rendering, etc... all being the implementations of the concepts.

I had watched a lot of talks about lit-html so I didn't feel like I really needed the extra info at the beginning—or that it was missing from the Writing templates section—but I did get to the concepts page and think, "oh... why is it giving me a really broad overview of the core concepts after I've read the most complex aspects of the library like the template syntax reference?

**re: nothing**
I think having the nothing part where it is does make sense, but I think the example makes too many assumptions about the readers knowledge. I think its fair for the lit-element documentation to make assumptions about the readers knowledge about lit-html, but maybe not the other way around. 

I think it might be more clear if you have a full example that uses plain html elements with the slot, and a full/complete example that someone can type in and run. Right now it's not clear how this example gets wired up. 


But again, these are just my thoughts. Take em' and do with them what you will. :) Overall I think these documents are great and I love the idea of making this even more approachable to people. I showed lit-html and jsx to a friend who is learning JS and he responded, "Why would anyone bother with JSX if they can just write JS in templates with lit-html?" LOL 

Lit-HTML really is superb. Thanks so much to all of your efforts!
